pipeline-templates/terraform.yaml

```diff
@@ -29,9 +29,18 @@
           workingDirectory: '$(System.DefaultWorkingDirectory)/components/${{ parameters.component }}'
           environmentServiceName: '${{ parameters.service_connection }}'
 
+      - task: Bash@3
+        condition: and(succeeded(), eq(variables['Build.SourceBranch'], 'refs/heads/main'))
+        name: "CheckFileChanges"
+        displayName: "CheckFileChanges"
+        env:
+          files_list: '${{ parameters.files_list }}'
+        inputs:
+          filePath: pipeline-scripts/check-changes.sh
+
       - task: TerraformCLI@0
         displayName: Apply - ${{ parameters.environment }} - ${{ parameters.component }}
-        condition: and(succeeded(), eq(variables.isMaster, true))
+        condition: or(and(succeeded(), eq(variables['CheckFileChanges.SOURCE_CODE_CHANGED'], 'true')),and(succeeded(), eq(${{ parameters.override }}, 'true')))
```

Comment:
can it be ?

condition: and(succeeded(), or(eq(variables['CheckFileChanges.SOURCE_CODE_CHANGED'], 'true'),eq(${{ parameters.override }}, 'true')))

Reply:
Good shout - now added